selectors/views/custom_status.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+import {Preferences} from 'mattermost-redux/constants';
+
+import {GlobalState} from 'types/store';
+
+export function getCustomStatus(state: GlobalState, userID?: string) {
```

Comment:
This selector needs to be made into a selector factory (like `makeGetCustomStatus`) since it takes an argument and returns new objects each time its called. Anywhere that we're passing its result into a component will cause that component to re-render a lot if we don't have some sort of memoization here, and since we'll be calling this with different user IDs, we'll need to memoize it on a per-component basis.